packages/xo-web/src/xo-app/vm/tab-advanced.js

```diff
@@ -288,6 +319,125 @@ class CoresPerSocket extends Component {
   }
 }
 
+class AddAclsModal extends Component {
+  get value() {
+    return this.state
+  }
+
+  _getPredicate = createSelector(
+    () => this.props.acls,
+    () => this.props.vm,
+    (acls, object) => ({ id: subject }) => !some(acls, { object, subject })
+  )
+
+  render() {
+    const { action, subjects } = this.state
+    return (
+      <form>
+        <div className='form-group'>
+          <SelectSubject
+            multi
+            onChange={this.linkState('subjects')}
+            predicate={this._getPredicate()}
+            value={subjects}
+          />
+        </div>
+        <div className='form-group'>
+          <SelectRole onChange={this.linkState('action')} value={action} />
+        </div>
+      </form>
+    )
+  }
+}
+
+const Acls = decorate([
+  addSubscriptions({
+    acls: subscribeAcls,
+    groups: cb => subscribeGroups(groups => cb(keyBy(groups, 'id'))),
+    users: cb => subscribeUsers(users => cb(keyBy(users, 'id'))),
+  }),
+  provideState({
+    effects: {
+      addAcls: () => (state, { acls, vm }) =>
+        confirm({
+          title: _('vmAddAcls'),
+          body: <AddAclsModal acls={acls} vm={vm} />,
+        })
+          .then(({ action, subjects }) => {
+            if (action == null || isEmpty(subjects)) {
+              throw new Error('User(s)/group(s) and role are required.')
+            }
+            return (
+              Promise.all(
+                map(subjects, subject =>
+                  addAcl({ subject, object: vm, action })
+                )
+              ),
+              noop
+            )
+          })
+          .catch(
+            err => err && error(_('addAclsError'), err.message || String(err))
+          ),
+      removeAcl: (_, { currentTarget: { dataset } }) => (_, { vm: object }) =>
+        removeAcl({
+          action: dataset.action,
+          object,
+          subject: dataset.subject,
+        }),
+    },
+    computed: {
+      rawAcls: (_, { acls, vm }) => filter(acls, { object: vm }),
+      resolvedAcls: ({ rawAcls }, props) =>
+        rawAcls.map(({ subject, ...acl }) => ({
+          ...acl,
+          subject: props.users[subject] || props.groups[subject],
+        })),
+    },
+  }),
+  injectState,
+  ({ state: { resolvedAcls }, effects, vm }) => (
+    <Container>
+      <Row>
+        {resolvedAcls.slice(0, 5).map(({ subject, action }) => (
+          <Col key={`${subject.email}.${action}`}>
```

Comment:
You're stacking 12 unit `Col`s which makes them look like `Row`s. Use `Row`s and `Col`s.